pkg/kvstore/lock.go

```diff
@@ -129,37 +130,40 @@ func LockPath(ctx context.Context, backend BackendOperations, path string) (l *L
 	lock, err := backend.LockPath(ctx, path)
 	if err != nil {
 		kvstoreLocks.unlock(path, id)
-		Trace("Failed to lock", err, logrus.Fields{fieldKey: path})
+		Trace(logger, "Failed to lock", fieldKey, path)
 		err = fmt.Errorf("error while locking path %s: %w", path, err)
 		return nil, err
 	}
 
-	Trace("Successful lock", err, logrus.Fields{fieldKey: path})
+	Trace(logger, "Successful lock", fieldKey, path)
 	return &Lock{kvLock: lock, path: path, id: id}, err
```

Comment:
Nit. Could save the logger in the `Lock` struct, so that it does not need to be explicitly passed to `Unlock`.